dashboard/src/shared/schema.ts

```diff
@@ -1,16 +1,74 @@
+// WARN: yaml doesn't have updated definitions for TypeScript
+// In particular, it doesn't contain definitions for `get` and `set`
+// that are used in this package
+import * as jsonSchema from "json-schema";
+import * as YAML from "yaml";
 import { IBasicFormParam } from "./types";
 
-export function retrieveBasicFormParams(schema: any) {
-  // TBD
-  return {
-    username: {
-      path: "wordpressUsername",
-      value: "user",
-    } as IBasicFormParam,
-  };
+// Avoid to explicitly add "null" when an element is not defined
+// tslint:disable-next-line
+const { nullOptions } = require("yaml/types");
+nullOptions.nullStr = "";
+
+// retrieveBasicFormParams iterates over a JSON Schema properties looking for `form` keys
+// It uses the raw yaml to setup default values.
+// It returns a key:value map for easier handling.
+export function retrieveBasicFormParams(
+  defaultValues: string,
+  schema: jsonSchema.JSONSchema4,
+): { [key: string]: IBasicFormParam } {
+  if (schema && schema.properties) {
+    return lookForFormParams(defaultValues, schema.properties);
+  }
+  return {};
```

Comment:
This function looks like it doesn't do much now - other than verify the schema has properties... why not rename `lookForFormParams` to `retrieveBasicFormParams` and have call-sites use that (if they have properties)? Given that `lookForPormParams` no longer has the array arg (which may have been awkward for call-sites to call with an empty array), it now looks exportable?

Reply:
yes, we can do that, I wanted to split it for being a bit more readable but probably there is not much difference.